compiler/src/main/java/org/qbicc/graph/CheckCast.java

```diff
@@ -10,7 +11,8 @@
  * This node is used to implement two distinct kinds of dynamic checks:
  * arraystorechecks and checkcasts.
  */
-public final class CheckCast extends AbstractValue implements CastValue {
+public final class CheckCast extends AbstractValue implements CastValue, OrderedNode {
+    private final Node dependency;
```

Comment:
I think you also need to add handling of `dependency` to `DotNodeVisitor.visit(final Appendable param, final CheckCast node)`
`

Reply:
Done